python-sdk/src/astro/databases/base.py

```diff
@@ -389,15 +389,18 @@ def create_schema_and_table_if_needed(
                 columns_names_capitalization=columns_names_capitalization,
             )
 
-    def fetch_all_rows(self, table: BaseTable) -> list:
+    def fetch_all_rows(self, table: BaseTable, row_limit: int = 0) -> list:
```

Comment:
`row_limit` on the function called `fetch_all_rows` doesn't make sense to me so I agree with @dimberman , the function is self explanatory and needed for internal usage only

Reply:
but that's fine, feel free to merge @dimberman